### DISCUSSION TITLE
Functions to plot BCs for vertex-grid using existing MF-files generated externally?

### DISCUSSION THREAD

**Discussion by d-strom** [modflowpy/flopy#2155] [Q&A] [answered]
Hi,

I'm looking for some script/functions to plot the recharge-array (amongst other BCs) for a modflow-6 model using vertex grid.

So far I have been able to plot the k-values for all layers using:

ml.npf.k.plot()

If i call the function for rch like below:

ml.npf.rch.plot()

I get an attribute error saying there is no 'rch' attribute.

I also saw an example where the function ml.rch.rech.plot(kper="all", masked_values=[0.0], colorbar=True) is used to plot the rch-array.

This gives me "AttributeError: 'ModflowGwfrch' object has no attribute 'rech'"

Can anyone advice me on how to plot it?

**Comment by wpbonelli** [comment, marked as answer]
> If i call the function for rch like below:

> ml.npf.rch.plot()

`flopy.mf6.ModflowGwfnpf` is the node property flow package, it does not have an `rch` attribute.

> I also saw an example where the function ml.rch.rech.plot(kper="all", masked_values=[0.0], colorbar=True) is used to plot the rch-array.

> This gives me "AttributeError: 'ModflowGwfrch' object has no attribute 'rech'"

Only `flopy.modflow.ModflowRch` (non-mf6) has a `rech` attribute.

To plot recharge for an mf6 model, you can use e.g. `PlotMapView.plot_bc("RCH")`, see `plot_grid()` [here](https://modflow6-examples.readthedocs.io/en/latest/_notebooks/ex-gwf-csub-p04.html#Plotting-results) for example. Or more generally, you can pass an array of data to `PlotMapView.plot_array()`, see `plot_recharge()` [here](https://modflow6-examples.readthedocs.io/en/latest/_notebooks/ex-gwf-nwt-p03.html#Plotting-results).

**Reply by d-strom** [reply]
Thanks! I'll look into it
